src/main/java/com/example/LoyaltyCard.java

```diff
@@ -0,0 +1,29 @@
+package com.example;
+
+import static com.google.common.base.Preconditions;
+
+
+/**
+ * Class implements Value Object.
```

Comment:
You don't need to refer to a patterns in Javadoc. It's usually assumed by the way a class is implemented. 

You can probably add such things using `@apiNote`, but it's too fine details for the current stage of implementation.

Please describe what the type is about in plain shop-related terms. Try formulating it on paper in Russian, and then translate.